src/components/widgets-settings/averages/averages.directive.coffee

```diff
@@ -0,0 +1,112 @@
+module = angular.module('impac.components.widgets-settings.averages',[])
+                .constant('GRANULARITY_OPTIONS', Object.freeze([
+                   {name: 'daily', value: 'DAILY'},
+                   {name: 'weekly', value: 'WEEKLY'},
+                   {name: 'monthly', value: 'MONTHLY'},
+                   {name: 'quarterly', value: 'QUARTERLY'},
+                   {name: 'yearly', value: 'YEARLY'}]))
+
+module.controller('SettingAveragesCtrl', ($scope, ImpacDashboardsSvc, GRANULARITY_OPTIONS) ->
+
+  w = $scope.parentWidget
+  $scope.granularityOptions = GRANULARITY_OPTIONS
+  $scope.seletedAverages =
+    enabled: false
+    period: 'DAILY'
+    calculation:
+      from: moment().subtract(6, 'months').toDate()
+      to: moment().toDate()
+    application:
+      from: moment().add(1, 'day').toDate()
+      to: moment().add(3, 'months').toDate()
+
+  $scope.toggleAverages = ->
+    $scope.seletedAverages.enabled = !$scope.seletedAverages.enabled
+
+  $scope.isToggleAverages = ->
+    if $scope.seletedAverages?
+      return $scope.seletedAverages.enabled
+
+  $scope.calculation =
```

Comment:
What do you think about DRYing this part a bit? We could do something like:
```javascript 
$scope.method = (type = $scope.seletedAverages.calculation) -> 
    from:
      datePicker:
        opened: false
        date: type.from
        options: {maxDate: new Date()}
        toggle: -> this.opened = !this.opened
    to:
      datePicker:
        opened: false
        date: type.to
        options: {minDate: new Date()}
        toggle: -> this.opened = !this.opened
```

Reply:
Good one! I'm doing it!